analyzere/base_resources.py

```diff
@@ -222,18 +222,28 @@ def upload_status(self):
     def upload_data(self, file_or_str):
         """
         Accepts a file-like object or string and uploads it. Files are
-        automatically uploaded in 4mb chunks. Implements the tus protocol.
+        automatically uploaded in chunks. The default chunk size is 16MiB and
+        can be overwritten by specifying the number of bytes in the
+        ``analyzere.upload_chunk_size`` variable.
+        Implements the tus protocol.
         """
         file_obj = StringIO(file_or_str) if isinstance(
             file_or_str, six.string_types) else file_or_str
-        chunk_size = 4 * 1024 * 1024
-        length = utils.file_length(file_obj)
 
-        # Initiate upload session
-        request_raw('post', self._data_path, headers={'Entity-Length': length})
+        # Upload file with known entity size if file object supports random
+        # access.
+        if hasattr(file_obj, 'seek'):
+            length = utils.file_length(file_obj)
+
+            # Initiate upload session
+            request_raw('post', self._data_path,
+                        headers={'Entity-Length': length})
+        else:
+            request_raw('post', self._data_path)
```

Comment:
To follow TUS protocol, we should eventually insert the entity-length (upload length) when we know it.  This is ok if it is on the last chunk.